irods_testing_environment/ssl_setup.py

```diff
@@ -84,11 +85,72 @@ def generate_ssl_dh_params(generator=2, key_size=1024, directory=None):
     return dhfile
 
 
-def configure_ssl_on_server(container,
-                            path_to_key_file_on_host,
-                            path_to_cert_file_on_host,
-                            path_to_dhparams_file_on_host):
-    """Copy SSL files to the container and configure SSL on the iRODS server.
+def configure_tls_for_service_account(container, cert_file):
+    """Configure TLS for the iRODS service account client environment.
+
+    Arguments:
+    container -- the docker.Container on which SSL will be configured
+    cert_file -- path to the file in the container containing the self-signed cert
+    """
+    cert_file = os.path.join(context.irods_config(), 'server.crt')
+
+    logging.warning(f'[{container.name}]: configuring SSL')
+
+    # add certificate chain file, certificate key file, and dh parameters file to iRODS
+    # service account environment file
+    service_account_irods_env = os.path.join(context.irods_home(), '.irods', 'irods_environment.json')
+    irods_env = json_utils.get_json_from_file(container, service_account_irods_env)
+
+    logging.debug(f'[{container.name}: env before [{json.dumps(irods_env)}]')
+
+    irods_env['irods_client_server_policy'] = 'CS_NEG_REQUIRE'
+    irods_env['irods_ssl_ca_certificate_file'] = cert_file
+    irods_env['irods_ssl_verify_server'] = 'cert'
+
+    logging.debug(f'[{container.name}: env after [{json.dumps(irods_env)}]')
+
+    json_utils.put_json_to_file(container, service_account_irods_env, irods_env)
+
+
+def configure_tls_in_server_config(container, key_file, cert_file, dhparams_file):
+    """Configure TLS in server_config on the iRODS server.
+
+    Arguments:
+    container -- the docker.Container on which SSL will be configured
+    key_file -- path to the file in the container containing the private key for the cert
+    cert_file -- path to the file in the container containing the self-signed cert
+    dhparams_file -- path to the file in the container containing the dhparams PEM file
+    """
+    from . import negotiation_key
+
+    key_file = os.path.join(context.irods_config(), 'server.key')
+    dhparams_file = os.path.join(context.irods_config(), 'dhparams.pem')
+    chain_file = os.path.join(context.irods_config(), 'chain.pem')
+    cert_file = os.path.join(context.irods_config(), 'server.crt')
```

Comment:
confused here... aren't these being passed in?

Reply:
Good catch. Don't need these.

Reply:
Done